raiden/blockchain/events.py

```diff
@@ -337,7 +337,8 @@ def poll_blockchain_events(self):
 
     def uninstall_all_event_listeners(self):
         for listener in self.event_listeners:
-            listener.filter.web3.eth.uninstallFilter(listener.filter.filter_id)
+            if listener.filter.filter_id:
+                listener.filter.web3.eth.uninstallFilter(listener.filter.filter_id)
```

Comment:
doesn't `web3.utils.filters.LogFilter` make it a noop if the `filter_id` is missing?

Reply:
No. According to their source, `web3.eth.uninstallFilter` performs no check on the id but immediately initiates an RPC call.

Source [here](https://github.com/ethereum/web3.py/blob/master/web3/eth.py#L357)